### ISSUE TITLE
app rejected from apple app store due to data stored in user's iCloud

### ISSUE THREAD

**Issue by tylordippel** [facebook/react-native#7395]
I'm using react-native 0.18.0. My app was rejected from the Apple app store with the following reason:
On launch and content download, your app stores 9.28MB on the user's iCloud, which does not comply with the iOS Data Storage Guidelines.

I am not intentionally syncing any data to iCloud. I suspect that the react-native javascript bundle is being unintentionally synced to iCloud. I saw a similar issue here: https://github.com/Microsoft/react-native-code-push/issues/320; however, I am not using CodePush. Where is the react-native javascript bundle stored on the phone? Could it be accidentally synced to iCloud? And if so, how do you avoid that?

**Comment by tylordippel** [comment]
I added the following line to my AppDelegate
`[jsCodeLocation setResourceValue:@YES forKey:NSURLIsExcludedFromBackupKey error:nil];`
Hopefully that fixes it! According to https://github.com/Microsoft/react-native-code-push/pull/321/files, this looks like it should work.

**Comment by lostintangent** [comment]
@tylordippel I can confirm that the mentioned fix addresses Apple's concern here. Of the many users we have with CodePush, we actually only saw app get rejected for this reason, but as soon as we explicitly disabled syncing for the assets that the CodePush plugin writes to disk, the user re-submitted the app to the store and was approved without any issues.

**Comment by tylordippel** [comment]
@lostintangent I re-submitted the app with the fix and unfortunately it was rejected again. I guess I'll have to take a look at my dependencies to make sure that they aren't synching to iCloud:

   "apsl-react-native-button": "^2.4.0",
    "base-64": "^0.1.0",
    "firebase": "^2.4.1",
    "react-native": "^0.18.0",
    "react-native-autolink": "^0.1.0",
    "react-native-linear-gradient": "^0.3.2",
    "react-native-modalbox": "^1.3.1",
    "react-native-scrollable-tab-view": "^0.3.4",
    "react-native-simple-store": "^0.1.0",
    "react-native-spinkit": "0.0.5",
    "react-native-swiper": "^1.4.3",
    "react-native-vector-icons": "^0.8.4",
    "react-native-video": "^0.6.1",
    "react-redux": "^3.1.0",
    "redux": "^3.0.5",
    "redux-thunk": "^1.0.3"

I am using a slightly old version of react-native. I'll try to update that as well.

Also, my app does retrieve some image data from my firebase db of the form 
**data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAASABI..** and then uses the built in react-native image component to display them. Could the image components be synching data do iCloud? The files in RCTImage.xcodeproj clearly use NSURL; however, I didn't see `NSURLIsExcludedFromBackupKey` anywhere in those files. Should it be there?

**Comment by cgreening** [comment]
Make sure you are not storing anything unecessary in the documents directory of the application. Or anything you are storing has the exclude from backup setting applied.

You can pull down your app's folder from the device using the devices window in XCode and see what is in the documents directory - or you can run it on the simulator and then browse to the relevant folder  in:

`~/Library/Application Support/iPhone Simulator`

**Comment by danleveille** [comment]
@tylordippel Did you figure out what the issue was here? I just got rejected for the same reason.

**Comment by tylordippel** [comment]
@danleveille Yes, I did; however, it was a very strange process. After my app was rejected the first time, I added `[jsCodeLocation setResourceValue:@YES forKey:NSURLIsExcludedFromBackupKey error:nil];` to my AppDelegate.m and resubmitted. For some reason, it was rejected a second time for the same reason. I then emailed the review team asking for a more detailed explanation and explained that I added the above line to prevent my js bundle from being backed up to iCloud. Then, a day later, my app went back into review and was approved. All they said was that my fix was sufficient. So adding the above line to your AppDelegate.m should be enough.
